payments-core/src/main/java/com/stripe/android/payments/PaymentFlowResultProcessor.kt

```diff
@@ -197,19 +196,34 @@ internal sealed class PaymentFlowResultProcessor<T : StripeIntent, out S : Strip
      * @throws MaxRetryReachedException when max retry is reached and the status is still
      * "requires_action".
      */
-    @Throws(
-        MaxRetryReachedException::class,
-        InvalidRequestException::class
-    )
     private suspend fun refreshStripeIntentUntilTerminalState(
         originalIntent: StripeIntent,
         clientSecret: String,
         requestOptions: ApiRequest.Options
-    ): T {
+    ): Result<T> {
         var remainingRetries = MAX_RETRIES
 
-        var stripeIntent = requireNotNull(
-            if (shouldCallRefreshIntent(originalIntent)) {
+        var stripeIntentResult = if (shouldCallRefreshIntent(originalIntent)) {
+            refreshStripeIntent(
+                clientSecret = clientSecret,
+                requestOptions = requestOptions,
+                expandFields = EXPAND_PAYMENT_METHOD
+            )
```

Comment:
I guess this is a small behavior change: Previously, any exception would bubble up to where `processResult()` is being called, such as `PaymentLauncherViewModel`. Now, we caught it here and schedule a retry. I feel like this new behavior is preferable, but let me know if you think otherwise.

Reply:
Added!